File: app/src/main/java/com/penseapp/acaocontabilidade/login/interactor/LoginInteractor.java
package com.penseapp.acaocontabilidade.login.interactor;

/**
 * Created by unity on 24/02/2016.
 */
public interface LoginInteractor {
    void signUp(String name, String user, String type, String password);
    void login(String email, String password);

    void getCurrentUser();
    void addAuthStateListener();
    void removeAuthStateListener();
}

File: app/src/main/java/com/penseapp/acaocontabilidade/login/interactor/LoginInteractor.kt
package com.penseapp.acaocontabilidade.login.interactor

/**
 * Created by unity on 24/02/2016.
 */
interface LoginInteractor {

    fun signUp(name: String, company: String, email: String, type: String, password: String)

    fun login(email: String, password: String)

    fun resetPassword(email: String)


    fun getCurrentUser()

    fun addAuthStateListener()

    fun removeAuthStateListener()
}
